packages/core/src/physics/CharacterController.ts

```diff
@@ -10,13 +10,30 @@ import { ColliderShape } from "./shape";
  * The character controllers.
  */
 export class CharacterController extends Collider {
-  private _stepOffset: number = 0.5;
+  private _stepOffset = 0.5;
   private _nonWalkableMode: ControllerNonWalkableMode = ControllerNonWalkableMode.PreventClimbing;
   private _upDirection = new Vector3(0, 1, 0);
-  private _slopeLimit: number = 0.707;
+  private _slopeLimit = 0.707;
+  private _contactOffset = 0.1;
+
+  /**
+   * Contact offset for this characterController, the value must be greater than or equal to 0.
+   */
+  get contactOffset(): number {
+    return this._contactOffset;
+  }
+
+  set contactOffset(value: number) {
+    value = Math.max(0, value);
+    if (this._contactOffset !== value) {
+      this._contactOffset = value;
+      (<ICharacterController>this._nativeCollider).setContactOffset(value);
+    }
+  }
 
   /**
    * The step offset for the controller.
+   * @remarks Character can overcome obstacle less than the height(stepOffset + contractOffset(0.1)).
```

Comment:
Maybe should modify when opt contractOffset

Reply:
done